api/models/config.go

```diff
@@ -125,3 +127,67 @@ func (h *Headers) Scan(value interface{}) error {
 	// otherwise, return an error
 	return fmt.Errorf("headers invalid db format: %T %T value, err: %v", value, bv, err)
 }
+
+// MilliCPU units
+type MilliCPUs uint64
+
+const (
+	MinMilliCPUs = 0       // 0 is unlimited
+	MaxMilliCPUs = 1024000 // 1024 CPUs
+)
+
+// implements fmt.Stringer
+func (c MilliCPUs) String() string {
+	if c == 0 {
+		return ""
+	}
+	return fmt.Sprintf("%dm", c)
+}
+
+// implements json.Unmarshaler
+func (c *MilliCPUs) UnmarshalJSON(data []byte) error {
+
+	outer := strings.TrimSpace(string(data))
+	if !strings.HasSuffix(outer, "\"") || !strings.HasPrefix(outer, "\"") {
+		return ErrInvalidJSON
+	}
+
+	outer = strings.TrimSuffix(outer, "\"")
+	outer = strings.TrimPrefix(outer, "\"")
+
+	CPUs := strings.TrimSpace(outer)
+	if CPUs == "" {
+		*c = 0
+		return nil
+	}
+
+	if strings.HasSuffix(CPUs, "m") {
+		// Support milli cores as "100m"
+		CPUs = strings.TrimSuffix(CPUs, "m")
+		mCPU, err := strconv.ParseUint(CPUs, 10, 64)
+		if err != nil || mCPU > MaxMilliCPUs || mCPU < MinMilliCPUs {
+			return ErrInvalidCPUs
+		}
+		*c = MilliCPUs(mCPU)
+	} else {
+		// Support for floating point "0.1" style CPU units
+		fCPU, err := strconv.ParseFloat(CPUs, 64)
+		if err != nil || fCPU < MinMilliCPUs/1000 || fCPU > MaxMilliCPUs/1000 {
+			return ErrInvalidCPUs
+		}
+		*c = MilliCPUs(fCPU * 1000)
+	}
+
+	return nil
+}
+
+// implements json.Marshaler
+func (c *MilliCPUs) MarshalJSON() ([]byte, error) {
+
+	if *c < MinMilliCPUs || *c > MaxMilliCPUs {
```

Comment:
kind of pedantic but this check here and in unmarshal seem like they might belong in `route.Validate()` since they're not parsing errors. this might get rid of the noted json handling woes

Reply:
i guess not quite on the json woes, since they could still not provide a valid float / millicpu value